internal/lagoon/compose.go

```diff
@@ -57,6 +59,9 @@ func UnmarshalLagoonDockerComposeYAML(file string) ([]OriginalServiceOrder, erro
 		// extract the services only
 		if item.Key.(string) == "services" {
 			for idx, v := range item.Value.(goyaml.MapSlice) {
+				if err := utilvalidation.IsDNS1035Label(v.Key.(string)); err != nil {
+					return nil, errors.New("service name cannot contain '.'")
```

Comment:
I think this will catch errors other than fullstops, e.g. being longer than 63 characters, starting with a digit etc. Perhaps a more generic error message "Service name is invalid. See the docs <link>" or something.

Reply:
I don't want hardcoded docs links in here, would get old real fast.

But generally clearer message referencing something that could exist in our docs that is searchable like "Read the documentation around service naming requirements" and then we get our documentation updated to mention this somewhere too

Reply:
Yep good call! I've updated the error message now, let me know if you want any changes

Reply:
Looks good to me, we probably need a followup PR in the main lagoon docs to mention this requirement